tethysext/atcore/services/workflow_manager/condor_workflow_manager.py

```diff
@@ -186,9 +186,13 @@ def prepare(self):
         # Save the workflow
         self.workflow.save()
 
-        # Preprocess jobs if they are dicts
-        if self.jobs_are_dicts:
-            self.jobs = self._build_job_nodes(self.jobs)
+        # Preprocess jobs if they are dicts or a callback function
+        cur_jobs = self.jobs(self) if self.jobs_are_function else self.jobs
+        self.validate_jobs(cur_jobs)  # Validate again (needed if self.jobs was a callback function)
```

Comment:
Refactor so we only call validate if we call the jobs function:

```python
if inspect.isfunction(self.jobs):
    cur_jobs = self.jobs(self)
    self.validate_jobs(cur_jobs)
else:
    cur_jobs = self.jobs
```

Reply:
Refactored as shown.